tripy/tripy/frontend/shape.py

```diff
@@ -160,4 +160,12 @@ def __radd__(self, other):
     def __eq__(self, other):
         from tripy.frontend.trace.ops.reduce import all
 
+        # `.shape` will contain a single int. To avoid endlessly calling
+        # Shape.__eq__, we can compare the int's directly using `.data()`
+        if self.shape.data() != other.shape.data():
```

Comment:
Not sure why is the return on line 168 required? When will that return False?

Reply:
168 would return False for something like `tp.Shape([1,2]) == tp.Shape([1,3])`. The check on 165 will check the actual rank. So `tp.Shape([1]) != tp.Shape([1,2])`. This case was missed in my previous MR. If we don't check the rank, Tripy would broadcast the inputs to the same shape, possibly causing `tp.Shape([1]) == tp.Shape([1,1])`.

Reply:
Oh I didn't read that `self.shape` is shape of shape tensor..

Reply:
Yes. Is there a way to do this that doesn't trigger lowering?

Reply:
You could also write this with `__len__` (just merged my change), though it could in some cases trigger lowering.